### PR TITLE
Routes enhancements

### PR DESCRIPTION
**Pull request by Harper04**
* added tests for route registration
* added route validation methods

this is a "living" branch; will push further changes and issue new pull request after this got merged

### CODE REVIEW

File: packages/routes/src/precedence.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const priorityB = pathPrecedence.map(test => test(b)).indexOf(true)
+  return priorityA === priorityB ? 0 : priorityA < priorityB ? -1 : 1
+}
+
+export const routeSort = (a: typeof Route, b: typeof Route) => {
+  const segmentPairs = pairsFromArrays(a.path.split('/'), b.path.split('/'))
+  const precedences = segmentPairs.map(pair => precedenceForSegment(...pair))
+  return precedences.find(p => p !== 0) || 0
+}
+
```
Comment (Aduril) [inline packages/routes/src/precedence.ts]: remove, because it is not necessary

Comment (DfHlyu4aida) [reply packages/routes/src/precedence.ts]: it is used in the tests

Comment (Harper04) [reply packages/routes/src/precedence.ts]: good point!

File: packages/routes/src/validation.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Route } from './Route'
+
+export const routeIsValid = <T extends typeof Route>(route: T) =>
+  pathIsValid(route.path)
+
```
[the remark this answers is not in the document]
Comment (Harper04) [reply packages/routes/src/validation.ts]: done

**Pull request merged**